src/server/auth/server/testing/auth_test.go

```diff
@@ -544,19 +544,34 @@ func TestCreateAndUpdatePipeline(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, "", infoAfter.AuthToken)
 
-	// And also check for a listPipeline
+	// Check that an unprivileged user can list pipelines without details
 	pipelineInfos, err := aliceClient.ListPipeline(false)
 	require.NoError(t, err)
+	require.Equal(t, 2, len(pipelineInfos))
 	for _, pipelineInfo := range pipelineInfos {
 		require.Equal(t, "", pipelineInfo.AuthToken)
 	}
-	// TODO(2.0 required): this call errors if it uses aliceClient (maybe skip
-	// pipelines we don't have read access on?)
-	// pipelineInfos, err = aliceClient.ListPipeline(true)
+
+	// Check that an unprivileged user can list pipelines with details,
+	// but the details are skipped
+	pipelineInfos, err = aliceClient.ListPipeline(true)
+	require.NoError(t, err)
+	require.Equal(t, 2, len(pipelineInfos))
+	for _, pipelineInfo := range pipelineInfos {
```

Comment:
I think it's actually bad to make the results kind of polymorphic like this, but I suppose we don't have any prior art in the codebase.  I feel like if we request details we should always return details and the correct behavior might be to filter out those pipelines?

For example, there is a function in `completions.go` - `PipelineCompletion` that will crash now.  Also, I believe the `PPS.ActivateAuth` RPC could crash `pachd` if called on a server with auth already active by an unprivileged user.

Reply:
I think from a product perspective omitting pipelines when a user doesn't have access is weird - it'll probably break the DAG in dash, for instance. I think we want to limit listing repos/pipelines at the project level, not the individual repo/pipeline level.

That said, dash also appears to rely on being able to access the details of the pipelines. So it kind of seems like the option here is to make spec commits world-readable, which would be consistent with the behaviour of 1.x.